app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java

```diff
@@ -174,6 +176,12 @@ protected void showStreamDialog(final StreamInfoItem item) {
                     StreamDialogEntry.share
             ));
         }
+        final boolean enableKodiEntry = KoreUtil.isServiceSupportedByKore(item.getServiceId())
+                && PreferenceManager.getDefaultSharedPreferences(context)
+                .getBoolean(context.getString(R.string.show_play_with_kodi_key), false);
```

Comment:
You could provide this as a method in `KoreUtil` so if the logic for when to show the kodi entry is centralized and not copied over 3 places.